modules/openid_auth/app/controllers/openid_auth/v2/validation_controller.rb

```diff
@@ -81,6 +86,28 @@ def setup_structure
         end
         payload_object
       end
+
+      def charon_token_screen?(payload_object)
+        act_vista_id = payload_object.act[:vista_id]
+        sta3n = payload_object.launch['sta3n']
+        return false unless !act_vista_id.nil? && !sta3n.nil?
+
+        vista_ids = act_vista_id.scan(/\d{3}[A-Z]*\|\d+\^[A-Z]{2}\^\d{3}[A-Z]*\^[A-Z]{5}\|[A-Z]{1}/)
+        return false unless vista_ids
+
+        vista_ids.each do |vista_id|
+          parsed_sta3n = vista_id.match(/\d{3}|/)
```

Comment:
Let me reach out to the team that provided me this regex and ask about valid `sta3n` values.

They didn't provide regex groupings, but assuming that `|` and `^` are delimiters, I'm wondering if this regex should actually be `\d{3}[A-Z]*`.

I'll let you know.

Reply:
I will change to make sure. It's safer to go with your recommendation for now and I'll adjust per feedback.

Reply:
The change I made doesn't include the last "\|" since so the parsed section wouldn't include that delimiter.